src/main/java/org/apache/sling/scriptingbundle/plugin/processor/ResourceTypeFolderAnalyser.java

```diff
@@ -66,6 +66,8 @@ public Capabilities getCapabilities(@NotNull Path resourceTypeDirectory) {
                                 fileProcessor.processExtendsFile(resourceType, entry, providedCapabilities, requiredCapabilities);
                             } else if (Constants.REQUIRES_FILE.equals(file.toString())) {
                                 fileProcessor.processRequiresFile(entry, requiredCapabilities);
+                            } else if (Constants.VAULT_CONTEXT_XML.equals(file.toString())) {
```

Comment:
Every docview file may be relevant, not only the ones named ".content.xml". You should check for extension .xml first and then you need to parse the first bytes to check if it is docview. Filevault provides a utility method for that.

Reply:
Do you have a link to this utility? Does it make sense in our context? We're only trying to figure out if we're in a component folder.

Reply:
https://github.com/apache/jackrabbit-filevault/blob/master/vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/XmlAnalyzer.java

Reply:
I discussed with @kwin in private and we said that for a first release of this feature we could work only with Generic Aggregates, which use the `.content.xml` file for serialization. We can add support for Full Coverage Aggregates later, if needed.

Reply:
In addition, it seems that the `XmlAnalyzer` has some issues in our current test setup - it only reads the first 1024 characters to figure out the serialization type. However, if the file contains a copyright header, then the class will fail to correctly identify the serialization type.